### PR TITLE
Create classes that add functions/parsers for each endpoint.

### PR DESCRIPTION
**Pull request by Mackey22**
* Rename `_get_endpoint_name` to `get_command_name` because it's public.
* Create the template for each auto-generated command class.
* Include each of the autogenerated command files.
* `Regenerate_api.py` basically takes functionality `parser.py` and dumps all parameter information for each command into the command files. Not a rename b/c tests rely on `parser.py` capability.

### CODE REVIEW

File: hca/added_command.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 import requests
 
-from constants import Constants
```
Comment (ttung) [inline hca/added_command.py]: Simple renames / refactors should be a different diff.

File: hca/regenerate_api.py [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        with open(test_api_path) as fp:
+            api_spec_dict = json.load(fp)
+    else:
+        api_spec_dict = requests.get("https://hca-dss.czi.technology/v1/swagger.json").json()
+    return api_spec_dict
+
+
+def _make_name(http_method, path_split):
+    """Name an endpoint."""
+    # If the api needs file/write/{sdf} functionality, will become put-file-write
```
Comment (ttung) [inline hca/regenerate_api.py]: is `sdf` just any generic noun?

File: hca/regenerate_api.py [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+    name = [http_method]
+    path_non_args = list(filter(lambda x: len(x) > 0 and x[0] != "{", path_split))
```
Comment (ttung) [inline hca/regenerate_api.py]: `list` not required

Comment (ttung) [inline hca/regenerate_api.py]: also, you can do `not x.startswith("{")`

Comment (ttung) [inline hca/regenerate_api.py]: finally, at this point, you might as well do:

```
for path_element in path_split:
  if path_element.startswith("{"):
    break
  name.append(path_element)
```

File: hca/regenerate_api.py [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+    """
+    For all positional arguments in an endpoint, determine if the argument is required.
+
+    If an argument is not always required (i.e. get-files/{uuid} and get-files are both
+    endpoints) then it will be labeled not required, but the endpoints it is required
+    will be marked to assist the user when using the help option.
+    """
+    # Find and clean all path args.
+    path_split = path.split("/")
+    path_args = list(filter(lambda x: len(x) > 0 and x[0] == "{", path_split))
```
Comment (ttung) [inline hca/regenerate_api.py]: see above.

File: hca/regenerate_api.py [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+            else:
+                options[param_name]['required'] = argrequired and options[param_name]['required']
+
+            # If the arg is required on this endpoint, add this endpoint to the list of
+            # endpoints for which this arg is needed.
+            if argrequired:
+                options[param_name]['required_for'].append(path)
+
+
+def get_parser(test_api_path=None):
```
Comment (ttung) [inline hca/regenerate_api.py]: does this name still reflect what this method does (does not look like it to me)?

File: hca/regenerate_api.py [hunk 6]
```diff
@@ -0,0 +1,372 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import json
+import os
+import shutil
+
+from jinja2 import FileSystemLoader, Environment
```
Comment (ttung) [inline hca/regenerate_api.py]: does this require a change to requirements.txt?

File: hca/added_command.py (hunk 1 above)
Comment (Mackey22) [inline hca/added_command.py]: Got it. Should I refactor this diff and make a new one for that or just keep that in mind for the next time?

File: hca/regenerate_api.py (hunk 6 above)
Comment (Mackey22) [inline hca/regenerate_api.py]: Yes, should be an update to setup.py. Will change.

File: hca/regenerate_api.py (hunk 2 above)
Comment (Mackey22) [inline hca/regenerate_api.py]: Yeah. Will replace with `generic`.

File: hca/regenerate_api.py (hunk 3 above)
Comment (Mackey22) [inline hca/regenerate_api.py]: 👍

File: hca/added_command.py (hunk 1 above)
Comment (ttung) [inline hca/added_command.py]: Eh. next time, I guess.

File: hca/api/autogenerated/get_bundles.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    def _get_base_url(cls):
+        return "https://hca-dss.czi.technology/v1"
+
+    @classmethod
+    def get_command_name(cls):
+        return "get-bundles"
+
+    @classmethod
+    def _get_endpoint_info(cls):
+        return {u'positional': [{u'description': u'Bundle unique ID.', u'format': None, u'pattern': u'[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}', u'required': False, u'argument': u'uuid', u'required_for': [u'/bundles/{uuid}'], u'type': u'string'}], u'seen': True, u'options': {u'version': {u'hierarchy': [u'version'], u'in': u'query', u'description': u'Timestamp of bundle creation in RFC3339.', u'required_for': [], u'format': u'date-time', u'pattern': None, u'array': False, u'required': False, u'type': u'string', u'metavar': None}, u'replica': {u'hierarchy': [u'replica'], u'in': u'query', u'description': u'Replica to fetch from.', u'required_for': [], u'format': None, u'pattern': None, u'array': False, u'required': False, u'type': u'string', u'metavar': None}}, u'description': u'Given a bundle UUID, return the latest version of that bundle.  If the version is provided, that version of the\nbundle is returned instead.\n'}
```
Comment (ttung) [inline hca/api/autogenerated/get_bundles.py]: can you tweak jinja to generate the trailing NL?

File: hca/api/templates/api.jinja [hunk 8]
```diff
@@ -0,0 +1,16 @@
+from ...added_command import AddedCommand
+
```
Comment (ttung) [inline hca/api/templates/api.jinja]: add a comment that is put into the autogenerated files telling people not to edit the autogenerated f iles.

File: hca/api/autogenerated/get_bundles.py (hunk 7 above)
Comment (Mackey22) [inline hca/api/autogenerated/get_bundles.py]: 👍

File: hca/api/templates/api.jinja (hunk 8 above)
Comment (Mackey22) [inline hca/api/templates/api.jinja]: 👍

**Comment by Mackey22** [comment]
Merging. Again, testing coming with next couple of prs.

**Pull request merged**